scout/parse/variant/transcript.py

```diff
@@ -96,42 +71,21 @@ def parse_transcripts(raw_transcripts, allele=None):
                 elif domain_name == "SMART_domains":
                     transcript["smart_domain"] = domain_id
 
-        coding_sequence_entry = entry.get("HGVSC", "").split(":")
-        protein_sequence_entry = entry.get("HGVSP", "").split(":")
-
-        coding_sequence_name = None
-        if len(coding_sequence_entry) > 1:
-            coding_sequence_name = coding_sequence_entry[-1]
-
-        transcript["coding_sequence_name"] = coding_sequence_name
-
-        protein_sequence_name = None
-        if len(protein_sequence_entry) > 1:
-            protein_sequence_name = protein_sequence_entry[-1]
-
-        transcript["protein_sequence_name"] = protein_sequence_name
-
+        transcript["coding_sequence_name"] = get_coding_sequence(entry)
+        transcript["protein_sequence_name"] = get_protein_sequence(entry)
         transcript["biotype"] = entry.get("BIOTYPE")
-
         transcript["exon"] = entry.get("EXON")
         transcript["intron"] = entry.get("INTRON")
-
-        if entry.get("STRAND"):
-            if entry["STRAND"] == "1":
-                transcript["strand"] = "+"
-            elif entry["STRAND"] == "-1":
-                transcript["strand"] = "-"
-        else:
-            transcript["strand"] = None
+        transcript["strand"] = get_strand(transcript, entry)
```

Comment:
Choose if you want to pass transcript, and update it, or take the return value! Its not currently used in the function.

Reply:
Yes. Will do.

Reply:
Done!